lineax/_operator.py

```diff
@@ -1429,6 +1447,14 @@ def _(operator):
     return diagonal, lower_diagonal, upper_diagonal
 
 
+@tridiagonal.register(DiagonalLinearOperator)
+def _(operator):
+    diagonal = jnp.diag(operator.as_matrix())
```

Comment:
This line could be `diagonal(operator)` to avoid materialising the whole matrix.